opal/mca/common/ucx/common_ucx_wpool.h

```diff
@@ -294,7 +294,7 @@ static inline int opal_common_ucx_wait_request_mt(ucs_status_ptr_t request, cons
                 break;
             }
             ctr--;
-        } while (ctr > 0 && ret > 0 && status == UCS_INPROGRESS);
+        } while (ctr > 0 && ret == 0 && status == UCS_INPROGRESS);
```

Comment:
Can you provide some context why this is necessary? My understanding is that previously we would progress UCX as long as there have been completions in UCX and otherwise call global progress. Now we would progress as long as there was no UCX completion? That means we would progress UCX much more even if the completion we're waiting for depends on progress in another part of OMPI... (eventually we'll jump to global progress once `progress_iterations` UCX progress calls were made but that seems like wasted time)

Reply:
Assuming that ret is 0 while ctr is still non-zero, we will never call opal_progress(). I found this hanging issue when running an application. Completion of this request depends on opal_progress call but we never called it and it hanged inside opal_common_ucx_wait_request_mt().

opal_common_ucx_wait_request_mt() is designed to avoid calling opal_progress() unnecessarily unless we have already progressed the worker enough times. This is a performance improvement as ucp_worker_progress() is way less expensive compared to opal_progress(). 

Depending on the application, progress_iterations can be tuned to gain higher performance.